kratos/utilities/constraint_utilities.cpp

```diff
@@ -66,6 +67,66 @@ void ComputeActiveDofs(
 /***********************************************************************************/
 /***********************************************************************************/
 
+void DistributedComputeActiveDofs(
+    ModelPart& rModelPart,
+    std::vector<int>& rActiveDofs,
+    const ModelPart::DofsArrayType& rDofSet,
+    const std::size_t InitialDofId
+    )
+{
+    KRATOS_TRY
+
+    // Get the data communicator
+    const auto& r_data_communicator = rModelPart.GetCommunicator().GetDataCommunicator();
+
+    // MPI data
+    const int rank = r_data_communicator.Rank();
+
+    // The number of local dofs
+    const std::size_t number_local_dofs = block_for_each<SumReduction<std::size_t>>(rDofSet, [&rank](const auto& rDof) {
+        if (rDof.GetSolutionStepValue(PARTITION_INDEX) == rank) {
+            return 1;
+        } else {
+            return 0;
+        }
+    });
+
+    // Base active dofs
+    rActiveDofs.resize(number_local_dofs);
+
+    block_for_each(
+        rActiveDofs,
+        [](int& r_dof)
+        { r_dof = 1; }
+    );
```

Comment:
Probably its better to use std::fill()as the compiler will likely replace it for a std::memset(), which cannot do if its inside the parallel lambda.

But I am not sure, we should measure it. 

(Same comment)

Reply:
The current one is faster, and if I try to use std::fill(std::execution::par, rActiveDofs.begin(), rActiveDofs.end(), 1); I have linklig problems with TBB (strange, I am not using TBB)

Reply:
Its used by the residual based builder and solve (as an option) if you compile with intel or MKL (or TBB). Otherwise windows is excruciatingly slow.

Reply:
Yes, but I am not invoking him, I am using std::execution::par, should use STL one. Any case I read in google that OMP is faster than std::fill without the execution policy, can we keep it as it is?

Reply:
Then we can keep it as it is